sql/hive/src/main/scala/org/apache/spark/sql/hive/TableReader.scala

```diff
@@ -283,7 +284,7 @@ class HadoopTableReader(
       _minSplitsPerRDD)
 
     // Only take the value (skip the key) because Hive works only with values.
-    rdd.map(_._2)
+    CombineSmallFile.combineWithPath(rdd, sc, path).map(_._2)
```

Comment:
@liancheng We can do this except `s3`, because if `your application needs to access large datasets, the fastest way to do that is to load them from Amazon S3 or an Amazon EBS device into an instance of the Hadoop Distributed File System (HDFS) on your nodes.` which write in `ec2-scripts.md`